compiler/src/dotty/tools/dotc/core/Types.scala

```diff
@@ -4272,7 +4263,7 @@ object Types {
           // constant if the term is constant
           case t: TermRef => isConst(t.underlying)
           // an operation type => recursively check all argument compositions
-          case applied: AppliedType if opsSet.contains(applied.typeSymbol.owner) =>
+          case applied: AppliedType if defn.isCompiletimeAppliedType(applied.typeSymbol) =>
```

Comment:
@soronpo, I found a regression after #13400 that I documented here: https://github.com/lampepfl/dotty/pull/14259#issuecomment-1013233452. It is due to the creation of the `opsSet`, more specifically to the `defn.*` calls. Using `defn.isCompiletimeAppliedType` instead solves the performance problem. Is it still correct? The tests pass.

Reply:
@mbovel Yes, you are correct. The regression was due to reconstruction of `opsSet` because it was not defined globally. Using `isCompiletimeAppliedType` is probably best, and we can get rid of `opsSet`.